bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/navigator/resources/ProjectExplorer.java

```diff
@@ -165,6 +166,7 @@ public void createPartControl(Composite aParent) {
 		if (this.userFilters.stream().anyMatch(UserFilter::isEnabled)) {
 			getCommonViewer().refresh();
 		}
+		getCommonViewer().setAutoExpandTrivialPathsLevel(AbstractTreeViewer.ALL_LEVELS);
```

Comment:
Enabling auto-expansion in a specific usage context should be done in a separate commit to providing the basic functionality.